### PR TITLE
14 compatibility

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: Christoph Wurst <christoph@winzerhof-wurst.at>

### CODE REVIEW

File: js/apps/Main.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import AppView from '../views/App';
 const Marionette = Backbone.Marionette;
 
 export default Marionette.Application.extend({
-	region: '#content',
+	region: '#app-content',
```
Comment (marcelklehr) [inline js/apps/Main.js]: this should probably be removed, as it's used by the AppView as a region.

File: js/apps/Main.js [hunk 2]
```diff
@@ -28,7 +28,8 @@ export default Marionette.Application.extend({
 		this.router = new Router({app: this});
 	},
 	onStart: function() {
-		this.showView(new AppView({app: this}));
+		var view = new AppView({app: this});
+		view.magic();
```
Comment (marcelklehr) [inline js/apps/Main.js]: Might be best to make the view accessible as a property.

File: js/views/App.js [hunk 3]
```diff
@@ -37,18 +36,23 @@ export default Marionette.View.extend({
 		}
 	},
 	initialize: function(options) {
+		this.bindUIElements();
```
Comment (marcelklehr) [inline js/views/App.js]: Where is this defined?

Comment (marcelklehr) [reply js/views/App.js]: Ah, that's from Marionette v2. We're using v3 ;)

File: css/bookmarks.css [hunk 4]
```diff
@@ -1,13 +1,3 @@
-#content > div {
-	height: 100%;
-	overflow: auto;
-}
-
-#app-navigation {
-	height: 100%;
-	position: relative;
```
Comment (marcelklehr) [inline css/bookmarks.css]: mmh. I'm pretty sure that was necessary for backwards compat to nextcloud v12 ^^
However, nextcloud GmbH drops support for that when the new release is out, right? So, we probably might as well stop supporting it.

Comment (skjnldsv) [reply css/bookmarks.css]: Changes for 14 are pretty big. Theoritecally, we need to check, this layout should be compatible with 13 too.
So yes, we need to drop 12. :)

**Comment by skjnldsv** [comment]
@marcelklehr thanks a lot for your help!! :)
I'm completely out of the loop with marionette. @ChristophWurst helped with his magic! ;)

**Comment by marcelklehr** [comment]
Do my changes work on v14? I don't have an instance at hand.

**Comment by skjnldsv** [comment]
@marcelklehr Yes, it looks fine here :)

**Pull request merged**